File: solutions/javascript/difference-of-squares/1/difference-of-squares.js
//
// This is only a SKELETON file for the 'Difference Of Squares' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Squares {
  constructor(number) {
    this.number = number;
  }

  get sumOfSquares() {
    let total = 0;

    let id = 1;
    while (id <= this.number) {
      total += (id * id);
    }

    return total;
  }

  get squareOfSum() {
    let total = 0;

    let id = 1;
    while (id <= this.number) {
      total += id;
    }

    return total * total;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}

File: solutions/c/difference-of-squares/1/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number) {
    unsigned int sum = 0;
    
    unsigned int i = 1;
    while (i <= number) {
        sum += i * i;
        i += 1;
    }
    
    return sum;
}

unsigned int square_of_sum(unsigned int number) {
    unsigned int sum = 0;

    unsigned int i = 1;
    while (i <= number) {
        sum += i;
        i += 1;
    }

    return sum * sum;
}

unsigned int difference_of_squares(unsigned int number) {
    return square_of_sum(number) - sum_of_squares(number);
}
